test/unit/test-amp-story-player.js

```diff
@@ -14,56 +14,126 @@
+      url =
+        'https://www-washingtonpost-com.cdn.ampproject.org/v/s/www.washingtonpost.com/graphics/2019/lifestyle/travel/amp-stories/a-locals-guide-to-what-to-eat-and-do-in-new-york-city/';
+      storyAnchor.setAttribute('href', url);
+      playerEl.appendChild(storyAnchor);
+    }
     win.document.body.appendChild(playerEl);
     manager = new AmpStoryPlayerManager(win);
   }
 
   beforeEach(() => {
     win = env.win;
-    buildStoryPlayer();
   });
 
   it('should build an iframe for each story', () => {
+    buildStoryPlayer();
     manager.loadPlayers();
 
     expect(playerEl.shadowRoot.querySelector('iframe')).to.exist;
   });
 
-  // TODO(#26308): unskip when messaging is enabled.
-  it.skip('should correctly append params at the end of the story url', () => {
+  it('should correctly append params at the end of the story url', () => {
+    buildStoryPlayer();
     manager.loadPlayers();
     const storyIframe = playerEl.shadowRoot.querySelector('iframe');
 
     expect(storyIframe.getAttribute('src')).to.equals(
-      url + '?amp_js_v=0.1#visibilityState=inactive&origin=about%3Asrcdoc'
+      url +
+        '?amp_js_v=0.1#visibilityState=visible&origin=about%3Asrcdoc&showStoryUrlInfo=0&storyPlayer=v0'
     );
   });
 
-  // TODO(#26308): unskip when messaging is enabled.
-  it.skip('should correctly append params at the end of a story url with existing params', () => {
+  it('should correctly append params at the end of a story url with existing params', () => {
+    buildStoryPlayer();
     url += '?testParam=true#myhash=hashValue';
     playerEl.firstElementChild.setAttribute('href', url);
 
     manager.loadPlayers();
     const storyIframe = playerEl.shadowRoot.querySelector('iframe');
 
     expect(storyIframe.getAttribute('src')).to.equals(
-      url + '&amp_js_v=0.1#visibilityState=inactive&origin=about%3Asrcdoc'
+      url +
+        '&amp_js_v=0.1#visibilityState=visible&origin=about%3Asrcdoc&showStoryUrlInfo=0&storyPlayer=v0'
+    );
+  });
+
+  it('should set first story as visible', () => {
+    buildStoryPlayer(3);
+    manager.loadPlayers();
+
+    const storyIframes = playerEl.shadowRoot.querySelectorAll('iframe');
+    expect(storyIframes[0].getAttribute('src')).to.include(
+      '#visibilityState=visible'
+    );
+  });
+
+  it('should prerender next stories', () => {
+    buildStoryPlayer(3);
+    manager.loadPlayers();
+
+    const storyIframes = playerEl.shadowRoot.querySelectorAll('iframe');
+    expect(storyIframes[1].getAttribute('src')).to.include(
+      '#visibilityState=prerender'
     );
   });
+
+  it(
+    'should remove iframe from a story with distance > 1 from current story ' +
+      'and give it to a new story that is distance <= 1 when navigating',
+    () => {
+      buildStoryPlayer(4);
+      const stories = toArray(playerEl.querySelectorAll('a'));
+
+      const player = new AmpStoryPlayer(win, playerEl);
+      player.buildCallback();
+      player.layoutCallback();
```

Comment:
Can we use `manager.loadPlayers()` here too?
If you need a way to retrieve the `player` instance to call `.next_()` until we can swipe, please add a TODO as well.